src/lib/chips/chip-list.ts

```diff
@@ -228,51 +443,26 @@ export class MdChipList implements AfterContentInit, OnDestroy {
   }
 
   /**
-   * Add a specific chip to our subscribed list. If the chip has
-   * already been subscribed, this ensures it is only subscribed
-   * once.
-   *
-   * @param chip The chip to be subscribed (or checked for existing
-   * subscription).
+   * Update key manager's active item when chip is deleted.
+   * If the deleted chip is the last chip in chip list, focus the new last chip.
+   * Otherwise focus the next chip in the list.
+   * Save `_lastDestroyedIndex` so we can set the correct focus.
    */
-  protected _addChip(chip: MdChip) {
-    // If we've already been subscribed to a parent, do nothing
-    if (this._chipSet.has(chip)) {
-      return;
-    }
-
-    // Watch for focus events outside of the keyboard navigation
-    chip._onFocus.subscribe(() => {
-      let chipIndex: number = this.chips.toArray().indexOf(chip);
-
-      if (this._isValidIndex(chipIndex)) {
-        this._keyManager.updateActiveItemIndex(chipIndex);
-      }
-    });
-
-    // On destroy, remove the item from our list, and setup our destroyed focus check
-    chip.destroy.subscribe(() => {
-      let chipIndex: number = this.chips.toArray().indexOf(chip);
-      if (this._isValidIndex(chipIndex)) {
-        if (chip._hasFocus) {
-          // Check whether the chip is the last item
-          if (chipIndex < this.chips.length - 1) {
-            this._keyManager.setActiveItem(chipIndex);
-          } else if (chipIndex - 1 >= 0) {
-            this._keyManager.setActiveItem(chipIndex - 1);
-          }
-        }
-        if (this._keyManager.activeItemIndex === chipIndex) {
-          this._lastDestroyedIndex = chipIndex;
+  protected _updateKeyManager(chip: MdChip) {
+    let chipIndex: number = this.chips.toArray().indexOf(chip);
+    if (this._isValidIndex(chipIndex)) {
+      if (chip._hasFocus) {
+        // Check whether the chip is the last item
+        if (chipIndex < this.chips.length - 1) {
```

Comment:
Nit: we could clarify this comment. The check is really for whether the chip is *not* the last item, right? I had to read it a few times before I realized it was doing the opposite